Dockerfile

```diff
@@ -36,8 +41,45 @@ RUN npm run webpack:build -- --env.SENTRY_DSN=$SENTRY_DSN
 WORKDIR /root/apps/site/react_renderer/
 RUN npm install && npx webpack
 
+# now, build the application back in the Elixir container
+FROM elixir-builder as app-builder
+
+WORKDIR /root
+
+# Add frontend assets compiled in the node container, required by phx.digest
+COPY --from=assets-builder /root/apps/site/priv/static ./apps/site/priv/static
+
 WORKDIR /root/apps/site/
+RUN mix do deps.compile
 RUN mix phx.digest
 
 WORKDIR /root
 RUN mix distillery.release --verbose
+
+
+# 3.) Use a debian container for the runtime environment
+FROM debian:buster
+
+# Set exposed ports
+EXPOSE 4000
+
+ENV PORT=4000 MIX_ENV=prod TERM=xterm LANG="C.UTF-8" REPLACE_OS_VARS=true
+
+# erlang-crypto requires system library libssl1.1
+RUN apt-get update && apt-get install -y --no-install-recommends \
+	libssl1.1 libsctp1 curl \
+	&& rm -rf /var/lib/apt/lists/*
+
+WORKDIR /root
+ADD . .
+
+COPY --from=app-builder /root/_build/prod/rel ./_build/prod/rel
+
+ADD rel/bin/startup /root/rel/site/bin/site/startup
```

Comment:
I think these two lines are where the permission error is coming from. `/root/rel/site/bin/startup` should be the path to the `startup` script, and it wants to find the `site` script in `/root/rel/site/bin/site`.

Reply:
```diff
diff --git a/Dockerfile b/Dockerfile
index aa85737d7..f27b6abdb 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -73,13 +73,13 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
 WORKDIR /root
 ADD . .

-COPY --from=app-builder /root/_build/prod/rel ./_build/prod/rel
+COPY --from=app-builder /root/_build/prod/rel /root/rel

-ADD rel/bin/startup /root/rel/site/bin/site/startup
+ADD rel/bin/startup /root/rel/site/bin/startup

 RUN mkdir /root/work

 WORKDIR /root/work

 # run the application
-CMD ["/root/rel/site/bin/site/startup", "foreground"]
+CMD ["/root/rel/site/bin/startup", "foreground"]
```

Reply:
Did you try running it locally? With these latest corrections I'm still getting the ol' `Error: No such container` error. Looking into it.

Reply:
It did, although I ran into a Node error:
```
2021-02-11T20:37:35.542 [error] node=site-unknown@2a955ad56cb3 Process #PID<0.2607.0> terminating
** (exit) exited in: Site.Application.start(:normal, [])
    ** (EXIT) shutdown: failed to start child: Site.React
        ** (EXIT) shutdown: failed to start child: :react_render
            ** (EXIT) an exception was raised:
                ** (MatchError) no match of right hand side value: {:error, {:enoent, [{:erlang, :open_port, [{:spawn_executable, nil}, [args: ["/root/rel/site/app.js"]]], []}, {Site.React.Worker, :init, 1, [file: 'lib/site/react/worker.ex', line: 15]}, {:gen_server, :init_it, 2, [file: 'gen_server.erl', line: 374]}, {:gen_server, :init_it, 6, [file: 'gen_server.erl', line: 342]}, {:proc_lib, :init_p_do_apply, 3, [file: 'proc_lib.erl', line: 249]}]}}
                    (poolboy 1.5.1) src/poolboy.erl:275: :poolboy.new_worker/1
                    (poolboy 1.5.1) src/poolboy.erl:296: :poolboy.prepopulate/3
                    (poolboy 1.5.1) src/poolboy.erl:145: :poolboy.init/3
                    (stdlib 3.12.1) gen_server.erl:374: :gen_server.init_it/2
                    (stdlib 3.12.1) gen_server.erl:342: :gen_server.init_it/6
                    (stdlib 3.12.1) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
    (kernel 6.5.2) application_master.erl:138: :application_master.init/4
    (stdlib 3.12.1) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Initial Call: :application_master.init/4
Ancestors: [#PID<0.2606.0>]
```